pkg/server/server.go

```diff
@@ -1849,12 +1849,13 @@ func (s *Server) PreStart(ctx context.Context) error {
 	// endpoints served by gwMux by the HTTP cookie authentication
 	// check.
 	if err := s.http.setupRoutes(ctx,
-		s.authentication,  /* authnServer */
-		s.adminAuthzCheck, /* adminAuthzCheck */
-		s.recorder,        /* metricSource */
-		s.runtime,         /* runtimeStatsSampler */
-		gwMux,             /* handleRequestsUnauthenticated */
-		s.debug,           /* handleDebugUnauthenticated */
+		s.rpcContext.TenantID, /* tenantID */
```

Comment:
In my PR I chose to have the `Flags` object as an argument and let `server.go` and `tenant.go` set whatever they want directly. What do you think of that alternative? https://github.com/cockroachdb/cockroach/pull/97945/files#diff-a28fa383e9ea02d78b07ec67d736a9c3e6a98a10de830399afa217d49b2ff618

Reply:
Good idea, this is much more flexible. I changed over to this approach based on your PR.